opennmt/utils/decoding.py

```diff
@@ -172,18 +172,24 @@ def initialize(self, batch_size, start_ids):
     initial_log_probs = tf.tile([0.] + [-float("inf")] * (self.beam_size - 1), [batch_size])
     parent_ids = tf.TensorArray(tf.int32, size=0, dynamic_size=True)
     maximum_lengths = tf.zeros([batch_size], dtype=tf.int32)
```

Comment:
`maximum_lengths` is no longer needed, can you remove it? In the `finalize` method it can be inferred from `sequence_lengths` with a `reduce_max` on the beam dimension.